cpp/src/arrow/engine/substrait/expression_internal.cc

```diff
@@ -951,6 +951,38 @@ struct ScalarToProtoImpl {
   Status Visit(const MonthIntervalScalar& s) { return NotImplemented(s); }
   Status Visit(const DayTimeIntervalScalar& s) { return NotImplemented(s); }
 
+  Status Visit(const Decimal32Scalar& s) {
```

Comment:
This is also a place where we'd benefit from a generic

Reply:
Converted to generic